api/service/staking/service.go

```diff
@@ -3,86 +3,148 @@ package staking
 import (
 	"crypto/ecdsa"
 	"math/big"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/params"
-	client "github.com/harmony-one/harmony/api/client/service"
+	protobuf "github.com/golang/protobuf/proto"
 	proto "github.com/harmony-one/harmony/api/client/service/proto"
 	"github.com/harmony-one/harmony/api/proto/message"
+	"github.com/harmony-one/harmony/core"
 	"github.com/harmony-one/harmony/core/types"
 	"github.com/harmony-one/harmony/internal/utils"
 	"github.com/harmony-one/harmony/p2p"
 )
 
+const (
+	// WaitTime is the delay time for resending staking transaction if the previous transaction did not get approved.
```

Comment:
in what condition the staking transaction will not get approved?

Reply:
staking transaction. in this PR, the new node will try forever to send staking transaction. I would add the logic of the newnode exitting after X attempts.